### PR TITLE
Add multiple response headers to open api

### LINKED ISSUE #8585: Support multiple headers in the response in the API definition
Currently there are multiple response headers for produceBlockV3:
Current definition in the beacon-api-spec (dev):

```
"headers": {
              "Eth-Consensus-Version": {
                "description": "Required in response so client can deserialize returned json or ssz data more effectively.",
                "required": true,
                "schema": {
                  "type": "string",
                  "enum": [
                    "phase0",
                    "altair",
                    "bellatrix",
                    "capella",
                    "deneb"
                  ],
                  "example": "phase0"
                }
              },
              "Eth-Execution-Payload-Blinded": {
                "description": "Required in response so client can deserialize returned json or ssz data to the correct object.",
                "required": true,
                "schema": {
                  "type": "boolean"
                }
              },
              "Eth-Execution-Payload-Value": {
                "description": "Execution payload value in Wei. Required in response so client can determine relative value\nof execution payloads.\n",
                "required": true,
                "schema": {
                  "type": "string",
                  "example": "1"
                }
              },
              "Eth-Consensus-Block-Value": {
                "description": "Consensus rewards paid to the proposer for this block, in Wei. Required in response so\nclient can determine relative value of consensus blocks.\n",
                "required": true,
                "schema": {
                  "type": "string",
                  "example": "1"
                }
              }
            },
```

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This PR adds support to multiple headers in the response, currently only used in the GetBlockV3 API.

GetBlockV3 render:
Screenshot 2024-09-13 at 1 53 49 PM

## Fixed Issue(s)

Fixes #8585 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/EthereumTypes.java [hunk 1]
```diff
@@ -99,6 +107,34 @@ public class EthereumTypes {
...
+  public static final StringTypeHeaderDefinition<UInt256> ETH_HEADER_EXECUTION_PAYLOAD_VALUE_TYPE =
+      new StringTypeHeaderDefinition.Builder<UInt256>()
+          .title(HEADER_EXECUTION_PAYLOAD_VALUE)
+          .description(
+              "Execution payload value in Wei. Required in response so client can determine relative value of execution payloads.")
+          .formatter(value -> value.toBigInteger().toString(10))
+          .parser(value -> UInt256.valueOf(new BigInteger(value, 10)))
+          .example("1")
+          .required(true)
+          .build();
```
Comment (rolfyone) [inline ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/EthereumTypes.java]: i think the name here may be inconsistent... 
```
StringBasedPrimitiveTypeDefinition extends StringValueTypeDefinition
DeserializableTypeDefinition
SerializableTypeDefinition
```
`StringTypeHeaderDefintion` is kind of a different pattern..

I wonder if we want something like 

`StringBasedHeaderTypeDefinition` and `BooleanHeaderTypeDefinition` or something...

File: infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/BooleanTypeHeaderDefinition.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  public BooleanTypeHeaderDefinition(
+      final String title, final Optional<Boolean> required, final String description) {
+    super(description);
+    this.title = title;
+    this.required = required;
+  }
+
+  @Override
+  public void serializeOpenApiType(final JsonGenerator gen) throws IOException {
```
Comment (rolfyone) [inline infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/BooleanTypeHeaderDefinition.java]: i think its worth building out some tests...

File: ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/EthereumTypes.java (hunk 1 above)
Comment (gfukushima) [reply ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/EthereumTypes.java:125]: truly the only difference is the way we serialize to the OpenAPI. But I could so with tiding up to follow a pattern.

Comment (gfukushima) [reply ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/EthereumTypes.java]: makes sense

**Review by rolfyone — APPROVED** [review]
LGTM!

**Pull request merged**